libretroshare/src/serialiser/rsserializer.h

```diff
@@ -226,7 +220,7 @@ struct RsGenericSerializer : RsSerialType
 		/** Allow shared allocator usage to avoid costly JSON deepcopy for
 		 *  nested RsSerializable */
 		SerializeContext(
-		        uint8_t *data, uint32_t size,
+		        uint8_t* data = nullptr, uint32_t size = 0,
```

Comment:
make sure that if only data is passed to SerializeContext, an error is printed out, otherwise you end up with some memory and no size for it.